pkg/services/sqlstore/searchstore/filters_test.go

```diff
@@ -0,0 +1,48 @@
+package searchstore_test
+
+import (
+	"testing"
+
+	"github.com/grafana/grafana/pkg/services/sqlstore/searchstore"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestFolderUIDFilter(t *testing.T) {
+	testCases := []struct {
+		description    string
+		uids           []string
+		expectedSql    string
+		expectedParams []interface{}
+	}{
+		{
+			description:    "searching general folder",
+			uids:           []string{"general"},
+			expectedSql:    "dashboard.folder_id = 0",
+			expectedParams: nil,
+		},
+
+		{
+			description:    "searching a specific folder",
+			uids:           []string{"abc-123"},
+			expectedSql:    "dashboard.folder_id IN (SELECT id FROM dashboard WHERE org_id = ? AND uid IN (?))",
+			expectedParams: []interface{}{int64(1), "abc-123"},
+		},
```

Comment:
Would it be worth adding the following test cases? 
(1) Including multiple UIDs in the array
(2) Including multiple UIDs in the array and one of them is the "general" folder. 

This is somewhat related to my previous comment so not sure if (2) is actually relevant.